File: src/server/src/main/java/com/project/application/models/post/CreatePostResource.java
package com.project.application.models.post;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nonnull;

import java.util.List;
import java.util.UUID;

@Serdeable
public record CreatePostResource(@Nonnull String text, @Nonnull String title, List<String> imageUrls, UUID categoryId) {
}

File: src/server/src/main/kotlin/com/project/application/models/post/CreatePostResource.kt
package com.project.application.models.post

import io.micronaut.serde.annotation.Serdeable
import jakarta.annotation.Nonnull
import java.util.UUID

@Serdeable
data class CreatePostResource(@Nonnull val text: String,
                         @Nonnull val title: String,
                         val previewText: String,
                         val imageUrls: List<String>,
                         val categoryId: UUID,
                         val tags: List<UUID>) {
}
